### ISSUE TITLE
Varia: Issues with button colors 

### ISSUE THREAD

**Issue by brookewp** [Automattic/themes#3365]
## Steps to replicate

If you activate Stratford and choose to use the theme's homepage, the issue seen in the screenshot below will be replicated. 

Some button colors aren't showing properly on Stratford. They look fine in the editor, but not on the live site. 

This issue can be seen on our current Stratford theme demo: https://stratforddemo.wordpress.com/
<img width="1585" alt="Screen Shot 2021-02-25 at 3 47 31 PM" src="https://user-images.githubusercontent.com/35543432/109238863-20d53280-7789-11eb-94d9-02cfa19044f7.png">

As well, there seems to be an issue with different states. I couldn't replicate this but have seen this in multiple reports (listed below). 

Normal state: 
<img width="460" alt="Screen Shot 2021-02-25 at 3 39 41 PM" src="https://user-images.githubusercontent.com/35543432/109238943-42361e80-7789-11eb-89d2-2bc86fe25bea.png">

Active state: 
<img width="441" alt="Screen Shot 2021-02-25 at 3 39 52 PM" src="https://user-images.githubusercontent.com/35543432/109238953-48c49600-7789-11eb-9d73-cb6c75be9bc1.png">

Visited state: 
<img width="591" alt="Screen Shot 2021-02-25 at 3 40 05 PM" src="https://user-images.githubusercontent.com/35543432/109238966-4d894a00-7789-11eb-8b33-2c47ba2c23f1.png">

## Result

Button isn't reflecting editor, active/hover/visited states are set to colors that cause visibility issues. 

## Expected

That the active, hover, and visited should be visible or there should be an option to set those separately. 

* User report: zd-3761048, hc-28020210

**Comment by nagpai** [comment]
Another issue here 
#28023027-hc

Workaround given is to choose the `Custom color` option and change the shade slightly by one digit/ alphabet.

**Comment by nagpai** [comment]
Maybe related 
https://github.com/Automattic/themes/pull/3338

**Comment by brookewp** [comment]
Edited title from 'Stratford' to 'Varia', as I saw this same issue on Rockfield hc-28098711

**Comment by pbking** [comment]
This doesn't appear to be breakage caused by #3251 (and related changes to other Varia children) which was my original guess as it is also happening prior to that change.  It reproduces all the way back to last year so it doesn't appear to be a recent change (at least for Stratford).

**Comment by katiebethbrown** [comment]
Another report in 28140925-hc
Theme: Barnsbury

Button in the editor:
<img src="https://user-images.githubusercontent.com/42279181/110400135-581ecb80-8034-11eb-8892-fdb325a523b8.png" width=250px>

Button on the site: 
<img src="https://user-images.githubusercontent.com/42279181/110400154-60770680-8034-11eb-894d-587f5554a784.png" width=250px>

Color on the site seems to be coming from this:
![image](https://user-images.githubusercontent.com/42279181/110400177-6b319b80-8034-11eb-8f72-87625197d92f.png)

**Comment by katiebethbrown** [comment]
Another report: 28326147-hc

**Comment by poojashetty18** [comment]
Another report: 29782261-hc
Theme: Barnsbury having the same button color issue as reported earlier

**Comment by Robertght** [comment]
Another case in 4156695-zen

The buttons become visible only on hover because the same color from the Customizer > Colors is being used to target the background of the same section: https://d.pr/i/J2hMaK

**Comment by ChrissiePollock** [comment]
Another case - 4380483-zen. 

The odd thing is that when I use my test site on the Business plan and set it up so it has the same theme and plugins as the person in that issue, then I copy the code from their site and try it on my page, it publishes properly. Screenshot in response to that customer.

**Comment by Nic-Sevic** [comment]
another case here with Barnsbury: 4435312-zd-woothemes
Seems also related this this issue which is marked closed?: https://github.com/Automattic/themes/issues/3250

**Comment by philnick206** [comment]
Another case here with Barnsbury: #4725327-zen

**Comment by noteeth** [comment]
Another report with Barnsbury #4881009-zen

**Comment by connorhipps** [comment]
Another report (?) with Exford: 4923221-zd-woothemes

**Comment by thekingsprojects** [comment]
Another possible case here: 5202545-zen

**Comment by supernovia** [comment]
I have a case here:
https://wordpress.com/forums/topic/text-on-buttons-not-showing/

**Comment by supernovia** [comment]
It seems to be this from _style.scss

```
	a.wp-block-file__button:active,
	a.wp-block-file__button:focus,
	a.wp-block-file__button:hover,
	a.wp-block-file__button:visited {
		color: #{map-deep-get($config-button, "color", "text-hover")};
		opacity: .85;
	}
```

combined with this from _config-child-theme-deep.scss

-- this is setting the text-hover to be the same color as the background.

```
/**
 * Button
 */
$config-button: (
	// Colors
	"color": (
		"text": map-deep-get($config-global, "color", "background", "default"),
		"text-hover": map-deep-get($config-global, "color", "background", "default"),
		"background": map-deep-get($config-global, "color", "primary", "default"),
		"background-hover": map-deep-get($config-global, "color", "primary", "hover"),
	),
```

**Comment by yashitamittal11** [comment]
I tested the Stratford theme and noticed two issues that are also present in the parent Varia theme.

1. Button text color changes to white on hover
2. If button background color = foreground color or background color and 
the user has selected custom text color, then
custom text color does not show.

This is the homepage of the Stratford demo theme-

![stratford](https://user-images.githubusercontent.com/21078987/183248719-b057dfc0-69fb-4a74-8d10-8e9b6fbc5cb4.png)

I modified the CSS in varia/sass/blocks/utilities/_style.scss file on my localhost to fix the 2nd issue.

This block of code is causing issues when the button's background color is set to the background color from the palette.


.has-background-background-color,
.has-background-background-color.has-background-dim {
	background-color: #{map-deep-get($config-global, "color", "background", "default")};
	color: #{map-deep-get($config-global, "color", "foreground", "default")};
}


I commented out the text color code as the text color is set by this CSS - 

.has-background-background-color:not(.has-text-color),
.has-background-background-color.has-background-dim:not(.has-text-color) {
	color: #{map-deep-get($config-global, "color", "foreground", "default")};
}

If the user selects a custom text color for the element, class **.has-text-color** is added to the element. If this class is added, we should not use the default color from the theme CSS.

__________________

This block of code is causing issues when the button's background color is set to the foreground color.

.has-background-dim,
.has-foreground-background-color,
.has-foreground-background-color.has-background-dim {
	color: #{map-deep-get($config-global, "color", "background", "default")};
}

Adding :not(.has-text-color) class fixes the issue.

![foreground-bg color](https://user-images.githubusercontent.com/21078987/183251631-4d0022d5-3649-4ffd-ae2b-30978675f851.png)

**Comment by yashitamittal11** [comment]
@madhusudhand Can you please share feedback on my approach to solving this issue?

**Comment by mikachan** [comment]
@yashitamittal11 Using `:not(.has-text-color)` looks like a good approach to me. I'd suggest opening up a PR with these changes 👍

**Comment by yashitamittal11** [comment]
>I tested the Stratford theme and noticed two issues that are also present in the parent Varia theme.

>1. Button text color changes to white on hover
>2. If button background color = foreground color or background color and
the user has selected custom text color, then
custom text color does not show.

This [PR](https://github.com/Automattic/themes/pull/6523) fixes the 2nd problem in the Varia theme. I am reopening this issue as the first problem requires a fix.

**Comment by happychait** [comment]
37153496-hc

**Comment by kavyagokul** [comment]
Reported in  #42611735-hc

I gave them a css workaround.

**Comment by jordesign** [comment]
No longer able to recreate this issue - so will close it.
